src/charm.py

```diff
@@ -745,6 +813,17 @@ def _on_config_changed(self, _):
                     msgs.append(f"Ingress IP(s): {', '.join(ingress_ips)}")
                 msgs.append(f"Service IP(s): {', '.join(self._report_service_ips())}")
                 msg = ", ".join(msgs)
+                self._delete_unused_services(svc_names)
+                svc_hostnames = [
```

Comment:
They are separated because in this way one function does just one thing and would be easier to modify/debug if needed. Another reason is that they use different API calls.

Reply:
What about adding another function that removes all the unused resources rather than writing that in the `_on_config_changed` code?

Reply:
Done.